locallibrary/catalog/forms.py

```diff
@@ -0,0 +1,47 @@
+import datetime
+
+from django import forms
+from django.core.exceptions import ValidationError
+from django.utils.translation import ugettext_lazy as _
+from django.forms import ModelForm
+from catalog.models import BookInstance
+
+class RenewBookForm(forms.Form):
+    renewal_date = forms.DateField(help_text="Enter a date between now and 4 weeks (default 3).")
+
+    def clean_renewal_date(self):
+        data = self.cleaned_data['renewal_date']
```

Comment:
việc get thuộc tính thông qua .cleaned_data có tác dụng gì em ?

Reply:
Get thuộc tính thông qua .cleaned_data có tác dụng làm cho dữ liệu được "làm sạch" và khử trùng Input mà nó có khả năng không an toàn bằng cách sử dụng trình xác thực mặc định và chuyển đổi thành loại tiêu chuẩn chính xác cho dữ liệu